### PR TITLE
chore: add analytics event for query failure post schema fetch pass

### LINKED ISSUE #28535: [Task]: Add analytics event to capture error when datasource schema fetch has passed but query fails
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [ref](https://www.notion.so/appsmith/5c8aa998aa94429995a6e0825ca2df05?v=14883d075b6649be9a8748938dcc224b&p=3c945dbbfed94978ae9fd65ce77fed30&pm=s)

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Add event to capture scenario where the schema fetch succeeds but data fetch fails for schema preview page. 
- Fix a bug where the error message and error type values were interchanged before the Appsmith plugin exception was returned to client. 

#### 
Fixes #28535 #28618

#### Media
Screenshot 2023-11-02 at 1 31 49 PM

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- Not aware how to add automation test for this at the moment.

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/hooks.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import type { DropdownOption } from "design-system-old";
 import { useDispatch, useSelector } from "react-redux";
 import { PluginPackageName } from "entities/Action";
 import { getCurrentEnvironmentId } from "@appsmith/selectors/environmentSelectors";
+import AnalyticsUtil from "../../../../../utils/AnalyticsUtil";
```
Comment (sneha122) [inline app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/hooks.ts:12]: @sumitsum We can simply import like `import AnalyticsUtil from "utils/AnalyticsUtil";`, no need to add exact path with ../../ etc. Please check DatasourceForm.tsx file for reference on the import

File: app/client/src/pages/Editor/DataSourceEditor/hooks.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { executeDatasourceQuery } from "actions/datasourceActions";
 import type { QueryTemplate } from "entities/Datasource";
 import { useState, useCallback } from "react";
 import { useDispatch } from "react-redux";
+import AnalyticsUtil from "../../../utils/AnalyticsUtil";
```
Comment (sneha122) [inline app/client/src/pages/Editor/DataSourceEditor/hooks.ts:5]: @sumitsum We can simply import like import AnalyticsUtil from "utils/AnalyticsUtil";, no need to add exact path with ../../ etc. Please check DatasourceForm.tsx file for reference on the import

File: app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/hooks.ts (hunk 1 above)
Comment (sumitsum) [reply app/client/src/pages/Editor/GeneratePage/components/GeneratePageForm/hooks.ts]: Done.

File: app/client/src/pages/Editor/DataSourceEditor/hooks.ts (hunk 2 above)
Comment (sumitsum) [reply app/client/src/pages/Editor/DataSourceEditor/hooks.ts]: Done.

**Review by sneha122 — APPROVED** [review]

**Pull request merged**